amundsen_application/api/utils/metadata_utils.py

```diff
@@ -74,6 +74,20 @@ def marshall_table_full(table_dict: Dict) -> Dict:
     return results
 
 
+def marshall_dashboard_full(dashboard_dict: Dict) -> Dict:
+    """
+    Cleanup some fields in the dashboard response
+    :param dashboard_dict: Dashboard response from metadata service.
+    :return: Dashboard dictionary with sanitized fields, particularly the tables and owners.
+    """
+    # TODO - Cleanup https://github.com/lyft/amundsen/issues/296
```

Comment:
I notice this ticket is closed though, can you reopen or file a new one with the information so your ideas don't get lost.

Reply:
I re-read the thread and I think this warrants further discussion. I have a half finished proposal doc about some of these fields. Either way I think this can be cleaned up at a later time. 

Some preliminary ideas about this:

1) Every resource should have a unique identifier. We can use the same field, `uri` for each, or use `uri` for dashboard, `key` for table and `user_id` for users. I prefer the former.

2) Deprecate `display_name` which is a generated field on FE Flask. In React, we could add a `getDisplayName(user)` function which will be something like
```
function getDisplayName(user) {
  if (!!user.full_name != null) return user.full_name;
  if (!!user.first_name && !!user.last_name) return ...
  else return user.email;
}
```

Reply:
Makes sense, also for the identifier the solid case for the same field is bookmarks. You will see in the PR  I push later -- with no changes we'd be stuck writing boolean logic for processing dashboard bookmarks with `uri` and table bookmarks with `key`. I had to introduce another temporary workaround in the python layer for this. 

I have thought on the 2nd item as well but to keep this PR focus on track that can be discussed later in your proposal.